### PR TITLE
refactor(http): Prepare "util/http" for queues

### PR DESCRIPTION
**Pull request by zharinov**
Prepare for #6983 merging

### CODE REVIEW

File: lib/util/http/index.ts [hunk 1]
```diff
@@ -53,21 +53,6 @@ function cloneResponse<T>(response: any): HttpResponse<T> {
...
-    const res = await promisedRes;
-    return cloneResponse(res);
-  } catch (err) {
-    if (abortOnError && !abortIgnoreStatusCodes?.includes(err.statusCode)) {
-      throw new ExternalHostError(err);
-    }
-    throw err;
-  }
-}
-
```
Comment (zharinov) [inline lib/util/http/index.ts:70]: Removed because it's inconvenient in terms of async/await linting

File: lib/util/http/index.ts [hunk 2]
```diff
@@ -80,6 +65,25 @@ function applyDefaultHeaders(options: Options): void {
...
+  httpRequests.push({
+    method: options.method,
+    url,
+    duration: responseTime - requestTime,
+    queueDuration: requestTime - startTime,
+  });
+  memCache.set('http-requests', httpRequests);
+  return resp;
+}
+
```
Comment (zharinov) [inline lib/util/http/index.ts:86]: Here we can place all the code related to `p-queue`

File: lib/workers/repository/stats.ts [hunk 3]
```diff
@@ -17,26 +22,35 @@ export function printRequestStats(): void {
...
+    const queueSum = requests
+      .map(({ queueDuration }) => queueDuration)
+      .reduce((a, b) => a + b, 0);
+    const queueAvg = Math.round(queueSum / hostRequests);
     const requestCount =
       `${hostRequests} ` + (hostRequests > 1 ? 'requests' : 'request');
-    hostStats.push(`${hostname}, ${requestCount}, ${avg}ms average`);
+    hostStats.push(
+      `${hostname}, ${requestCount}, ${requestAvg}ms request average, ${queueAvg}ms queue average`
+    );
```
Comment (zharinov) [inline lib/workers/repository/stats.ts:53]: Maybe we need check for `queueAvg === 0`, or just revert `stats.ts` for now

Comment (rarkins) [reply lib/workers/repository/stats.ts:53]: It's ok to have it temporarily as 0ms

**Review by rarkins — CHANGES_REQUESTED** [review]
Needs deconflicting

**Pull request merged**